File: Web/client/src/app/interfaces/game.ts
import { GameDifficulty } from '../enums/game-difficulty';
import { GameModes } from '../enums/game-modes';
import { GameStates } from '../enums/game-states';
import { PlayerPositions } from '../enums/player-positions';
import { PlayerStates } from '../enums/player-states';

export interface Game {
  id: string;
  host: string;
  currentImageId: string;
  mode: GameModes;
  state: GameStates;
  difficulty: GameDifficulty;
  round: number;
  start: Date;
  nextEvent: Date;
  players: Map<string, PlayerPositions>;
  roles: Map<string, PlayerStates>;
  scores: Map<string, number>;
  users: string[];
}

File: Remise/ClientLeger/polydessinmobile/app/src/main/java/com/example/polydessinmobile/Game.kt
package com.example.polydessinmobile

import java.util.*

data class Game(val difficulty: String = "",
                val mode: String = "",
                val round: Int = 0,
                val scores: Map<String, Int> = mapOf("player1" to 0),
                val players: Map<String, Int> = mapOf("player1" to 0),
                val roles: Map<String, String> = mapOf("player1" to "no role"),
                val nextEvent: Date = Date(System.currentTimeMillis()),
                val host: String = "",
                val start: Date = Date(System.currentTimeMillis()),
                val state: Int = 0,
                val currentImageId: String = ""
)